### PR TITLE
fix(react): Change to useLayoutEffect in useSnapshot

### PR DESCRIPTION
**Pull request by endash**
Address spurious consistency check re-renders by using useLayoutEffect inside useSnapshot instead of useEffect

## Related Issues or Discussions

Fixes #890

## Summary

Changing from `useEffect` to `useLayoutEffect` fixes a timing issue in `useSnapshot` where a React consistency check runs before the effect to store the cached snapshot, causing a spurious re-render. 

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Comment by endash** [comment]
That makes complete sense. The existing basic test I worked from verifies
the core orthogonality guarantee, these just guard against a performance
regression.

File: tests/optimization.test.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { useState } from 'react'
+import { fireEvent, render, waitFor } from '@testing-library/react'
+import { expect, it, vi } from 'vitest'
+import { proxy, snapshot, useSnapshot } from 'valtio'
```
Comment (dai-shi) [inline tests/optimization.test.tsx:4]:
```suggestion
import { proxy, useSnapshot } from 'valtio'
```

**Review by dai-shi — APPROVED** [review]
Looks great to me. Thanks for your contribution!

**Comment by endash** [comment]
🙌

**Pull request merged**